src/bun.js/http.exports.js

```diff
@@ -1,12 +1,199 @@
+    return Infinity;
+  }
+
+  constructor(options = kEmptyObject) {
+    super();
+    this.#options = options = { ...options, path: null };
+    if (options.noDelay === undefined) options.noDelay = true;
+
+    // Don't confuse net and make it think that we're connecting to a pipe
+    this.#requests = kEmptyObject;
+    this.#sockets = kEmptyObject;
+    this.#freeSockets = kEmptyObject;
+
+    this.#keepAliveMsecs = options.keepAliveMsecs || 1000;
+    this.#keepAlive = options.keepAlive || false;
+    this.#maxSockets = options.maxSockets || Agent.defaultMaxSockets;
+    this.#maxFreeSockets = options.maxFreeSockets || 256;
+    this.#scheduling = options.scheduling || "lifo";
+    this.#maxTotalSockets = options.maxTotalSockets;
+    this.#totalSocketCount = 0;
+  }
+
+  get defaultPort() {
+    return this.#defaultPort;
+  }
+
+  get protocol() {
+    return this.#protocol;
+  }
+
+  get requests() {
+    return this.#requests;
+  }
+
+  get sockets() {
+    return this.#sockets;
+  }
+
+  get freeSockets() {
+    return this.#freeSockets;
+  }
+
+  get options() {
+    return this.#options;
+  }
+
+  get keepAliveMsecs() {
+    return this.#keepAliveMsecs;
+  }
+
+  get keepAlive() {
+    return this.#keepAlive;
+  }
+
+  get maxSockets() {
+    return this.#maxSockets;
+  }
+
+  get maxFreeSockets() {
+    return this.#maxFreeSockets;
+  }
+
+  get scheduling() {
+    return this.#scheduling;
+  }
+
+  get maxTotalSockets() {
+    return this.#maxTotalSockets;
+  }
+
+  get totalSocketCount() {
+    return this.#totalSocketCount;
+  }
+
+  createConnection() {
+    debug(
+      `${NODE_HTTP_WARNING}\n`,
+      "WARN: Agent.createConnection is a no-op, returns fake socket",
+    );
+    return (this.#fakeSocket ??= new FakeSocket());
+  }
+
+  getName(options = kEmptyObject) {
+    let name = `http:${options.host || "localhost"}:`;
+    if (options.port) name += options.port;
+    name += ":";
+    if (options.localAddress) name += options.localAddress;
+    // Pacify parallel/test-http-agent-getname by only appending
+    // the ':' when options.family is set.
+    if (options.family === 4 || options.family === 6)
+      name += `:${options.family}`;
+    if (options.socketPath) name += `:${options.socketPath}`;
+    return name;
+  }
+
+  addRequest() {
+    debug(`${NODE_HTTP_WARNING}\n`, "WARN: Agent.addRequest is a no-op");
+  }
+
+  createSocket(req, options, cb) {
+    debug(
+      `${NODE_HTTP_WARNING}\n`,
+      "WARN: Agent.createSocket returns fake socket",
+    );
+    cb(null, (this.#fakeSocket ??= new FakeSocket()));
+  }
+
+  removeSocket() {
+    debug(`${NODE_HTTP_WARNING}\n`, "WARN: Agent.removeSocket is a no-op");
+  }
+
+  keepSocketAlive() {
+    debug(`${NODE_HTTP_WARNING}\n`, "WARN: Agent.keepSocketAlive is a no-op");
+  }
+
+  reuseSocket() {
+    debug(`${NODE_HTTP_WARNING}\n`, "WARN: Agent.reuseSocket is a no-op");
+  }
+
+  destroy() {
+    debug(`${NODE_HTTP_WARNING}\n`, "WARN: Agent.destroy is a no-op");
```

Comment:
Did you check these functions don't expect to return this?

Reply:
Yes none of the Agent functions actually return this, I think because they are mostly used internally from within `http.request` or `http.Server`, so they aren't really expected to be chained